plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/TypeEmbedding.kt

```diff
@@ -6,14 +6,20 @@
 package org.jetbrains.kotlin.formver.embeddings
 
 import org.jetbrains.kotlin.formver.conversion.SpecialFields
-import org.jetbrains.kotlin.formver.viper.domains.NullableDomain
-import org.jetbrains.kotlin.formver.viper.domains.UnitDomain
+import org.jetbrains.kotlin.formver.domains.NullableDomain
+import org.jetbrains.kotlin.formver.domains.TypeDomain
+import org.jetbrains.kotlin.formver.domains.UnitDomain
 import org.jetbrains.kotlin.formver.viper.ast.Exp
 import org.jetbrains.kotlin.formver.viper.ast.PermExp
 import org.jetbrains.kotlin.formver.viper.ast.Type
 
 interface TypeEmbedding {
-    val type: Type
+    // type represents a Viper expression specifying the type. They are defined in TypingDomain and are used for casting, subtyping and the is operator.
+    val kotlinType: Exp
+
+    // This represents the Viper type that an expression of this type has (e.g. an object is embedded as a Viper Ref type)
+    val embeddingType: Type
```

Comment:
I find this name a bit confusing, since I would expect this to denote a `TypeEmbedding`; perhaps `viperType` is better?

Reply:
I though about naming it `viperType` but then you could come into a situation where you have `typeEmbedding.viperType.toViper()` which is a bit awkward. But looking at the code this actually never occurs so except you have another even better name, I'll rename it to `viperType`.